src/include/function/hash/hash_functions.h

```diff
@@ -117,5 +117,13 @@ inline void Hash::operation(const std::unordered_set<std::string>& key, common::
     }
 }
 
+struct InternalIDHasher {
```

Comment:
I think this class in only used by unordered_set when computing hash for internalID and it is only used in path_function_executor. Maybe move to the path_function_executor.h?

Reply:
Also used for recursive join. So refactored into common